app/Http/Controllers/TopicController.php

```diff
@@ -46,7 +47,29 @@ private function getMetaDataForModel(Topic $topic)
         ];
 
         return $data;
+    }
+
+    /**
+     * @param Topic $topic
+     * @return array
+     */
+    private function getLikesOfTopic(Topic $topic)
+    {
+        $topic->countOfLikes = $topic->likes()->count();
+
+        $like = $topic->likes()->where('user_id', Auth::user()->id)->first();
+        if(!empty($topic->likes()->where('user_id', Auth::user()->id)->get()->first()))
+        {
+            $topic->is_user = true;
+            $topic->like_id = $topic->likes()->where('user_id', Auth::user()->id)->get()->first()->id;
+        }
```

Comment:
would be better

 if(!empty($like = $topic->likes()->where('user_id', Auth::user()->id)->get()->first()))
        {
            $topic->is_user = true;
            $topic->like_id = $like->id;
        }